fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java

```diff
@@ -187,17 +185,17 @@ private Partition getPartition(Long recordCount,
         for (int i = 0; i < partitionValues.length; i++) {
             String column = partitionColumnNames.get(i);
             String value = partitionValues[i].trim();
-            if (partitionColumnTypes.get(i) instanceof DateType) {
+            if (partitionColumnTypes.get(i) instanceof DateType && !Boolean.parseBoolean(partitionLegacyName)) {
```

Comment:
`partitionLegacyName` is a parameter from the caller, shall not be parsed again and again in a for loop.